### PR TITLE
Add wireshark extcap support

### PR DESCRIPTION
**Pull request by gpotter2**
This PR adds support for [wireshark extcap](https://www.wireshark.org/docs/man-pages/extcap.html). It works on both linux and windows (note: extcap uses fifo on Linux and NamedPipes on Windows so it was a bit annoying on Windows).

This is pretty cool because it adds support for a ton of **weird hardware**.

I tested the PR on:
- [`nrf sniffer`](https://www.nordicsemi.com/Products/Development-tools/nrf-sniffer-for-bluetooth-le) (a bluetooth dongle stick)
- [usbpcap](https://desowin.org/usbpcap/)
- [`sshdump`](https://www.wireshark.org/docs/man-pages/sshdump.html)

and it worked surprisingly okay.

### CODE REVIEW

File: scapy/layers/bluetooth4LE.py [hunk 1]
```diff
@@ -115,6 +121,84 @@ class BTLE_RF(Packet):
...
+    """
+    fields_desc = [
+        ByteField("board_id", 0),
+        LenField("len", None, fmt="<H", adjust=lambda x: x + 6),
+        ByteField("version", 2),
+        LEShortField("counter", None),
+        ByteEnumField("type", None, {
+            0x00: "req_follow",
+            0x01: "event_follow",
+            0x02: "event_device",             # missing from spreadsheet
```
Comment (guedou) [inline scapy/layers/bluetooth4LE.py:136]: What spreadsheet is that?

File: scapy/libs/extcap.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    """
+    p = subprocess.Popen(
+        [prog] + args,
+        stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+        text=True
+    )
+    data, err = p.communicate()
+    if p.returncode != 0:
+        raise OSError("%s returned with error code %s: %s" % (prog, p.returncode, err))
+    res = collections.defaultdict(list)
```
Comment (guedou) [inline scapy/libs/extcap.py:61]: Could we use a namedtuple here? That could help improve code readability.

File: scapy/libs/extcap.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        fifo, fd = _create_fifo()
+        args = ["--extcap-interface", iface, "--capture", "--fifo", fifo] + args
+        self.proc = subprocess.Popen(
+            [cmdprog] + args,
+        )
+        self.fd = _open_fifo(fd)
+        self.reader = PcapReader(self.fd)  # type: ignore
+        self.ins = self.reader  # type: ignore
+
+    def recv(self, x: int = MTU) -> Packet:
```
Comment (guedou) [inline scapy/libs/extcap.py:148]: Shall we raise an exception when using `def send()`?

**Review by guedou — CHANGES_REQUESTED** [review]
That's super cool! Did you try it with sshdump?

File: scapy/libs/extcap.py (hunk 2 above)
Comment (gpotter2) [reply scapy/libs/extcap.py:61]: I tried for a bit and I find it more confusing.
I didn't find a clean way of iterating over the named values of a namedtuple so it makes it pretty cumbersome here.

File: scapy/layers/bluetooth4LE.py (hunk 1 above)
Comment (gpotter2) [reply scapy/layers/bluetooth4LE.py]: Updated. See the top of the file.

File: scapy/libs/extcap.py (hunk 3 above)
Comment (gpotter2) [reply scapy/libs/extcap.py:161]: Shouldn't be necessary: l3socket or l2socket can't be built (only l2sniff).

**Comment by gpotter2** [comment]
> Did you try it with sshdump?

Just did and it's working fine !

**Comment by guedou** [comment]
Excellent !

**Review by guedou — APPROVED** [review]

**Pull request merged**